File: Proyecto-PracticasProfesionales-main-corregido/src/domain/configuracion/ConfiguracionSistema.ts
export interface ConfiguracionSistema {
  id_configuracion?: number;
  nombre_sistema: string;
  escuela_facultad: string;
  correo_institucional: string;
  estado_sistema: string;
  inscripcion_empresas_estado: string;
  inscripcion_empresas_motivo?: string | null;
  ciclo_escolar: string;
  hero_titulo: string;
  hero_subtitulo: string;
  id_convocatoria_principal: number | null;
  convocatoria_nombre: string;
  convocatoria_inicio: string | null;
  convocatoria_cierre: string | null;
  convocatoria_empresas_inicio?: string | null;
  convocatoria_empresas_cierre?: string | null;
  convocatoria_periodo?: string | null;
  convocatoria_estado?: string | null;
  soporte_telefono: string | null;
  ultima_actualizacion?: string | null;
}

File: Proyecto-PracticasProfesionales-main-corregido/backend/interfaces/api/schemas/configuracion_sistema.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, EmailStr


class ConfiguracionSistemaUpdate(BaseModel):
    nombre_sistema: str
    escuela_facultad: str
    correo_institucional: EmailStr
    estado_sistema: str
    inscripcion_empresas_estado: str = "Abierta"
    ciclo_escolar: str
    hero_titulo: str
    hero_subtitulo: str
    id_convocatoria_principal: int | None = None
    soporte_telefono: str | None = None


class ConfiguracionSistemaResponse(ConfiguracionSistemaUpdate):
    id_configuracion: int
    inscripcion_empresas_motivo: str | None = None
    convocatoria_nombre: str
    convocatoria_inicio: date | None = None
    convocatoria_cierre: date | None = None
    convocatoria_empresas_inicio: date | None = None
    convocatoria_empresas_cierre: date | None = None
    convocatoria_periodo: str | None = None
    convocatoria_estado: str | None = None
    ultima_actualizacion: datetime | None = None

    model_config = ConfigDict(from_attributes=True)
